packages/product-components/src/components/EditableText/EditableText.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { ReactNode, useEffect } from 'react';
+
+import styled, { css } from 'styled-components';
+
+import { borders, Elevation, mapElevationToBorder, nextElevation } from '@trezor/theme';
+import {
+    useElevation,
+    TransientProps,
+    FrameProps,
+    FramePropsKeys,
+    pickAndPrepareFrameProps,
+    withFrameProps,
+} from '@trezor/components';
+
+import { useOuterClick, useShortcuts } from './utils';
+import { ActionsContainer } from './ActionsContainer';
+
+export const allowedEditableTextFrameProps = [
+    'margin',
+    'maxWidth',
+] as const satisfies FramePropsKeys[];
+type AllowedFrameProps = Pick<FrameProps, (typeof allowedEditableTextFrameProps)[number]>;
+
+export type EditableTextProps = {
+    children: React.ReactNode;
+    onSave: (value: string) => void;
+    onFocus?: () => void;
+    onBlur?: () => void;
+    isLoading?: boolean;
+    isDisabled?: boolean;
+    isDeleteButtonVisible?: boolean;
+} & AllowedFrameProps;
+
+const BORDER_SIZE = 6;
+
+const EditableContainer = styled.span<TransientProps<AllowedFrameProps>>`
+    white-space: nowrap;
+    overflow: auto;
+    display: inline-flex;
+
+    ${withFrameProps}
+
+    &::-webkit-scrollbar {
+        display: none;
+    }
+`;
+
+const Container = styled.span<{ $elevation: Elevation; $isEditable: boolean }>`
+    position: relative;
+
+    ${({ $isEditable, $elevation, theme }) =>
+        $isEditable &&
+        css`
+            cursor: ${$isEditable ? 'text' : 'inherit'};
+            &::before {
+                content: '';
+                position: absolute;
+                top: -${BORDER_SIZE}px;
```

Comment:
These four lines can be replaced with just `inset: -6px`, no?

Reply:
done